cli/src/cmd/create.rs

```diff
@@ -0,0 +1,199 @@
+            eyre::bail!("could not find artifact")
+        }
+
+        Ok(())
+    }
+}
+
+impl CreateArgs {
+    /// Find using only ContractName
+    fn get_artifact_from_name(
+        &self,
+        compiled: ProjectCompileOutput<MinimalCombinedArtifacts>,
+    ) -> Result<(Contract, BytecodeObject)> {
+        let mut has_found_contract = false;
+        let mut contract_artifact = None;
+
+        for (name, artifact) in compiled.into_artifacts() {
+            let artifact_contract_name = name.split(':').collect::<Vec<_>>()[1];
+
+            if artifact_contract_name == self.contract.name {
+                if has_found_contract {
+                    eyre::bail!("contract with duplicate name. pass path")
+                }
+                has_found_contract = true;
+                contract_artifact = Some(artifact);
+            }
+        }
+
+        Ok(match contract_artifact {
+            Some(artifact) => (
+                artifact.abi.ok_or_else(|| eyre::Error::msg("message"))?,
+                artifact.bin.ok_or_else(|| eyre::Error::msg("message"))?,
+            ),
+            None => {
+                eyre::bail!("could not find artifact")
+            }
+        })
+    }
+
+    /// Find using src/ContractSource.sol:ContractName
+    fn get_artifact_from_path(
+        &self,
+        project: &Project,
+        path: String,
+    ) -> Result<(Contract, BytecodeObject)> {
+        // Get sources from the requested location
+        let abs_path = std::fs::canonicalize(PathBuf::from(path))?;
+        let mut sources = Sources::new();
+        sources.insert(abs_path.clone(), Source::read(&abs_path)?);
+
+        // Get artifact from the contract name and sources
+        let mut config = SolFilesCache::builder().insert_files(sources.clone(), None)?;
+        config.files.entry(abs_path).and_modify(|f| f.artifacts = vec![self.contract.name.clone()]);
+
+        // let binding
+        let _artifacts =
+            config.read_artifacts::<MinimalCombinedArtifacts>(project.artifacts_path())?;
+
+        let artifacts = _artifacts.values().collect::<Vec<_>>();
+
+        if artifacts.is_empty() {
+            eyre::bail!("could not find artifact")
+        } else if artifacts.len() > 1 {
+            eyre::bail!("duplicate contract name in the same source file")
+        }
+
+        Ok((
+            artifacts[0].clone().abi.ok_or_else(|| eyre::Error::msg("message"))?,
+            artifacts[0].clone().bin.ok_or_else(|| eyre::Error::msg("message"))?,
+        ))
+    }
+
+    fn deploy(
+        self,
+        abi: Contract,
+        bin: BytecodeObject,
+        args: Option<String>,
+        signer: SignerMiddleware<Provider<Http>, impl Signer + 'static>,
+    ) -> Result<()> {
+        let deployer_address = signer.address();
+        let factory = ContractFactory::new(abi, bin.as_bytes().unwrap().clone(), Arc::new(signer));
+
+        let deployer = match args {
+            Some(args) => factory.deploy(args)?,
```

Comment:
Don't think this will work, we'll need to parse the args (which should be a `Vec<String>` instead of just `String`) into a `Vec<Token>`, similar to this functionality: https://github.com/gakonst/foundry/blob/master/utils/src/lib.rs#L119-L125.

So I'd roll in the `build_constructor_with_args` logic in this function basically

Reply:
`deploy_tokens` works great. deployed a few contracts with and without constructor arguments